Source/ARTRealtime.m

```diff
@@ -547,70 +601,28 @@ - (ARTEventListener *)performTransitionWithStateChange:(ARTConnectionStateChange
             
             // RTN15g We want to enforce a new connection also when there hasn't been activity for longer than (idle interval + TTL)
             if (stateChange.previous == ARTRealtimeDisconnected || stateChange.previous == ARTRealtimeSuspended) {
-                NSTimeInterval intervalSinceLast = [[NSDate date] timeIntervalSinceDate:_lastActivity];
-                if (intervalSinceLast > (_maxIdleInterval + _connectionStateTtl)) {
-                    [self.connection setId:nil];
-                    [self.connection setKey:nil];
-                    [self.connection setSerial:0];
-                }
+                [self resetInactiveConnection];
             }
             
+            if (!_transport) {
```

Comment:
It creates timer which will (or will not) fire later than any other lines in this method. So it's position at the end of this code block kinda makes sense and seems doesn't affect anything (but if you insist I'll change it back).

Reply:
https://github.com/ably/ably-cocoa/pull/1691/commits/268df6605ea48b6052ac0e65016dd39294caad2e

Reply:
Did you think about it from the point of view of a reviewer, though? It's not clear whether the change was intentional or  not, and there's nothing in the commit message to explain the motivation.

Reply:
And it's part of a commit which already has quite a few changes, so for the purpose of reviewability, it would probably be best to avoid making unnecessary changes in the same commit.

Reply:
https://github.com/ably/ably-cocoa/pull/1691/commits/daaa89fc82b5e68226463de7081c1cab929d885c